src/NuGet.Clients/NuGet.CommandLine/Commands/RestoreCommand.cs

```diff
@@ -258,22 +264,26 @@ public RestoreCommand()
                 // Resolve the packages directory
                 Console.LogVerbose($"Using packages directory: {request.PackagesDirectory}");
 
-                if (externalProjects != null)
+                if (projectReferences != null)
                 {
-                    foreach (var externalReference in externalProjects)
+                    foreach (var projectReference in projectReferences)
                     {
-                        var projectDir = Path.GetDirectoryName(externalReference);
-                        var childProjectName = Path.GetFileNameWithoutExtension(externalReference);
-                        var childProjectJson =
-                            BuildIntegratedProjectUtility.GetProjectConfigPath(projectDir, childProjectName);
-
-                        Debug.Assert(childProjectJson != null && File.Exists(childProjectJson), childProjectJson);
-
-                        request.ExternalProjects.Add(
-                            new ExternalProjectReference(
-                                externalReference,
-                                childProjectJson,
-                                projectReferences: Enumerable.Empty<string>()));
+                        var projectDir = Path.GetDirectoryName(projectReference);
+                        var projectReferenceName = Path.GetFileNameWithoutExtension(projectReference);
+
+                        var childProjectJson = BuildIntegratedProjectUtility.GetProjectConfigPath(
+                            projectDir,
+                            projectReferenceName);
+
```

Comment:
Yup. The method returns projectname.project.json (if it exists) or project.json.